scripts/IVIndex.py

```diff
@@ -5,221 +5,208 @@
-            for text in text_content:
-                # tokenized_words is the list of terms in the block
-                raw_words = re.findall(r'\b\w+\b', text["plain_text"])
-
-                if rich:
-                    # annotations is a set containing annotations applied to the text.
-                    # It is empty if just regular plain text, options are the following:
-                    # bold, italic, strikethrough, underlined, code
-                    annotations = set(map(lambda x: x[0], filter(lambda x: (
-                        x[0] != 'color') and x[1], text["annotations"].items())))
-
-                    # type is the notion block type. e.g. text, heading1, callout, etc.
-                    type = text['type']
-                    for word in raw_words:
-                        token = tokenize_word(word)
-                        if token:
-                            terms.append(((token, (type, annotations))))
-                else:
-                    for word in raw_words:
-                        token = tokenize_word(word)
-                        if token:
-                            terms.append((token))
-    return terms
-
-
-def construct_page_id_lexicon(page_ids):
-    i = 0
-    lexicon = {}
-    for page_id in page_ids:
-        if not page_id in lexicon:
-            lexicon[page_id] = i
-            i += 1
-    return lexicon
-
-
-def construct_term_id_lexicon(terms):
-    i = 0
-    lexicon = {}
-    for term in terms:
-        if not term in lexicon:
-            lexicon[term] = i
-            i += 1
-    return lexicon
-
-
-def term_frequencies(terms):
-    freqs = defaultdict(int)
-    for term in terms:
-        freqs[term] += 1
-    return freqs
-
-def document_length(page_blocks):
-    count = 0
-    for block in page_blocks:
-        block_type = block["type"]
-        block_content = block[block_type]
-        if "rich_text" in block_content:
-            text_content = block_content["rich_text"]
-            for text in text_content:
-                # tokenized_words is the list of terms in the block
-                raw_words = re.findall(r'\b\w+\b', text["plain_text"])
-                for word in raw_words:
-                    token = tokenize_word(word)
-                    if token:
-                        count +=1
-    return count
-
-
-def get_page_lengths(pages):
-    """
-    Returns BM25 score for a term, and a Document
-
-    ripped straight from wikipedia
-    
-    Parameters:
-        - pages: dict[page_id] = { page blocks[str] } 
-    Returns:
-        - page_lengths: dict[page_id] -> number of tokens in page
-    """
-    
-    page_lengths = {}
-
-    for page_id, blocks in pages.items():
-        page_lengths[page_id] = document_length(blocks)
+class IVIndex: 
+    def __init__(self, page_ids, notion_secret, rich=False) -> None:
+        self.page_ids = page_ids
+        self.notion_secret = notion_secret
+        self.notion_pages = self.get_notion_pages(page_ids)
```

Comment:
alot of things don't need need to be stateful because they are tracked by NotionSearch but whatever